modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-types/FieldTypeList.es.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import classNames from 'classnames';
+import {
+	EVENT_TYPES as CORE_EVENT_TYPES,
+	useConfig,
+	useForm,
+	useFormState,
+} from 'dynamic-data-mapping-form-renderer';
+import React from 'react';
+
+import {dropLayoutBuilderField} from '../../../js/actions.es';
+import CollapsablePanel from '../../../js/components/collapsable-panel/CollapsablePanel.es';
+import EmptyState from '../../../js/components/empty-state/EmptyState.es';
+import FieldType from '../../../js/components/field-types/FieldType.es';
+import {getSearchRegex} from '../../../js/utils/search.es';
+
+const FieldTypeWrapper = ({
+	expanded,
+	fieldType,
+	fieldTypes,
+	showArrows,
+	...otherProps
+}) => {
+	const dispatch = useForm();
+	const {activePage, pages} = useFormState();
+
+	const getIcon = () => {
+		if (showArrows) {
+			return expanded ? 'angle-down' : 'angle-right';
+		}
+
+		return fieldType.icon;
+	};
```

Comment:
Could We move this function to be outside the component?

Reply:
Yes we can, but doing this we will must add parameters to the function, adding extra complexity to understand what it does. 
What do you think, @diegonvs?